### PR TITLE
Adapt SwDD for authorization

### PR DESCRIPTION
**Pull request by christoph-hamm**
Issues: #

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: agent/doc/swdesign/README.md
Comment (inf17101) [inline agent/doc/swdesign/README.md:1]: I suggest to describe the `Authorizer` object in the design decisions at the top of this README and then using the tag `-Authorizer` in the swdds.

Next, the swdd `swdd~agent-compares-control-interface-metadata~1`, which I have created within the refactoring of ControlInterface PR #345, must be extended to include the comparison of the `Authorizer`s (it is compared when updating a workload according to `swdd~agent-workload-obj-update-command~1`)

Comment (christoph-hamm) [reply agent/doc/swdesign/README.md:1]: I have made the requested changes.

Comment (krucod3) [reply agent/doc/swdesign/README.md:1]: @inf17101, what is the status here?

Comment (inf17101) [reply agent/doc/swdesign/README.md:1]: Changes are fine for me.

File: agent/doc/swdesign/README.md [hunk 1]
```diff
@@ -2892,17 +2900,68 @@ Needs:
...
 - utest
 
-#### Agent forwards Control Interface request fom the pipe to the server
-`swdd~agent-forward-request-from-control-interface-pipe-to-server~1`
+#### Agent checks Control Interface request for authorization
+`swdd~agent-checks-request-for-authorization~1`
+
+Status: approved
+
+When the Ankaios Agents receives a Control Interface request from a Workload, the Ankaios Agent shall check if this Workload is allowed to make this request.
```
Comment (inf17101) [inline agent/doc/swdesign/README.md:2908]:
```suggestion
When the Ankaios Agent receives a Control Interface request from a Workload, the Ankaios Agent shall check if this Workload is allowed to make this request.
```

File: agent/doc/swdesign/README.md [hunk 2]
```diff
@@ -2892,17 +2900,68 @@ Needs:
...
+- impl
+- utest
+
+#### Error returned on denied Control Interface request contains requst ID
+`swdd~agent-responses-to-denied-request-from-control-interface-contains-request-id~1`
 
 Status: approved
 
-When the Ankaios Agents receives a Control Interface request from a Workload, the Ankaios Agent shall forward this request to the Ankaios Server.
+When the Ankaios Agents sends a denied request error response to workload,
```
Comment (inf17101) [inline agent/doc/swdesign/README.md:2941]:
```suggestion
When the Ankaios Agent sends a denied request error response to workload,
```

File: agent/doc/swdesign/README.md [hunk 3]
```diff
@@ -3011,6 +3070,144 @@ Needs:
...
+Needs:
+- impl
+- utest
+
+#### Matching of deny rules
+`swdd~agent-authorizing-matching-deny-rules~1`
+
+Status: approved
+
+When the Ankaios Agent checks if an individual entry of the update/field mask of an request matches an individual entry of the filter mask of an deny rule, the Ankaios Agent shall consider them matching if all segments of the allow rule's filter mask match the corresponding segments of the request's update/field mask.
```
Comment (inf17101) [inline agent/doc/swdesign/README.md:3177]:
```suggestion
When the Ankaios Agent checks if an individual entry of the update/field mask of a request matches an individual entry of the filter mask of a deny rule, the Ankaios Agent shall consider them matching if all segments of the allow rule's filter mask match the corresponding segments of the request's update/field mask.
```

File: agent/doc/swdesign/README.md [hunk 4]
```diff
@@ -3011,6 +3070,144 @@ Needs:
...
+Needs:
+- impl
+- utest
+
+#### Matching of allow rules
+`swdd~agent-authorizing-matching-allow-rules~1`
+
+Status: approved
+
+When the Ankaios Agent checks if an individual entry of the update/field mask of an request matches an individual entry of the filter mask of an allow rule, the Ankaios Agent shall consider them matching if all segments of the allow rule's filter mask match the corresponding segments of the request's update/field mask.
```
Comment (inf17101) [inline agent/doc/swdesign/README.md:3158]:
```suggestion
When the Ankaios Agent checks if an individual entry of the update/field mask of a request matches an individual entry of the filter mask of an allow rule, the Ankaios Agent shall consider them matching if all segments of the allow rule's filter mask match the corresponding segments of the request's update/field mask.
```

File: agent/doc/swdesign/README.md [hunk 5]
```diff
@@ -3011,6 +3070,144 @@ Needs:
...
+Needs:
+- impl
+- utest
+
+#### Matching of rule elements
+`swdd~agent-authorizing-matching-rules-elements~1`
+
+Status: approved
+
+When the Ankaios Agent checks if one segment of an individual entry of the update/field mask of an request matches on segment an individual entry of the filter mask of an deny rule,
```
Comment (inf17101) [inline agent/doc/swdesign/README.md:3196]:
```suggestion
When the Ankaios Agent checks if one segment of an individual entry of the update/field mask of an request matches on segment an individual entry of the filter mask of a deny rule,
```

Isn't it for allow and deny rule?

**Review by inf17101 — COMMENTED** [review]
First part finished.

**Review by inf17101 — COMMENTED** [review]
Review finished.

File: tests/resources/image/Cargo.toml [hunk 6]
```diff
@@ -1,7 +1,9 @@
 [package]
 name = "control_interface_tester"
+version = "0.4.0"
```
[the remark this answers is not in the document]
Comment (krucod3) [reply tests/resources/image/Cargo.toml:3]: This is the version of the Control Interface Tester app itself and it does not have any effect at all. The released container is currently at `manual-build-3` and we planned to make the releases automated, but for now they are just incremental. If you ask me, we should omit the version here to reduce confusion:
```suggestion
```

**Review by inf17101 — APPROVED** [review]
LGTM 👍

**Pull request merged**